packages/chat-client/src/App.tsx

```diff
@@ -1,49 +1,88 @@
-import { useEffect, useState } from 'react'
-import { DeepChat } from "deep-chat-react";
-import './App.css'
+import { useEffect, useState } from 'react';
+import ChatWidget from './components/ChatWidget';
+import './App.css';
+import { setupIframeDetection } from './iframeDetection';
 
 function App() {
-
   const [modelId, setModelId] = useState<string | null>(null);
-  const [errorMessage, setErrorMessage] = useState<string | null>(null);
-  
-  // fetch modelId parameter from URL
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  // Fetch modelId from URL parameters
   useEffect(() => {
-    const urlParams = new URLSearchParams(window.location.search);
-    const modelId = urlParams.get('modelId');
-    if (modelId) {
-      fetch(`${import.meta.env.VITE_BACKEND_BASE_URL}/deepchat-proxy/${modelId}`)
-        .then(response => response.json())
-        .then(data => {
-          if (data) {
-            setModelId(modelId);
-          }
-        })
-        .catch(_error => {
-          setErrorMessage('Error while fetching modelId');
-        });
-    }
-  }, [modelId]);
-
-  // return error message if modelId is not provided
-  if (!modelId) {
+    const fetchModelData = async () => {
+      try {
+        setIsLoading(true);
+        const urlParams = new URLSearchParams(window.location.search);
+        const modelIdParam = urlParams.get('modelId');
+        const isEmbed = urlParams.get('embed') === 'true';
+
+        // Set embed mode in a state or context if needed
+        if (isEmbed) {
+          document.documentElement.classList.add('in-chat-iframe');
```

Comment:
Directly modifying the document in react is odd, especially since we are using MUI theming. We can better handling theming through MUI rather then this approach.